purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1019,57 +1026,31 @@ class Purchases internal constructor(
                             completion
                         )
                     } else {
-                        if (isBC5Enabled) {
-                            getProductDetails(subscriptionIds, { productsById ->
-                                val offerings = offeringsJSON.createOfferings(productsById)
-
-                                logMissingProducts(offerings, productsById)
-
-                                if (offerings.all.isEmpty()) {
-                                    handleErrorFetchingOfferings(
-                                        PurchasesError(
-                                            PurchasesErrorCode.ConfigurationError,
-                                            OfferingStrings.CONFIGURATION_ERROR_PRODUCTS_NOT_FOUND
-                                        ),
-                                        completion
-                                    )
-                                } else {
-                                    synchronized(this@Purchases) {
-                                        deviceCache.cacheOfferings(offerings)
-                                    }
-                                    dispatch {
-                                        completion?.onReceived(offerings)
-                                    }
+                        getSKUDetails(isBC5Enabled, subscriptionIds, { productsById ->
+                            val offerings = offeringsJSON.createOfferings(productsById)
+
+                            logMissingProducts(offerings, productsById)
```

Comment:
One thing that needs to change here is the assumption that products will map 1:1 to SKUs and we can keep them in a map.

I think we should pull in the mapping offering.packages to StoreProducts down into the BillingImplementation (because it'll depend on whether it's BC$ just by SKU, or BC5 which will use duration).
If we then return StoreProducts that are tagged by an offeringPackageIdentifier Purchases can still do a general mapping (although it's also not 1:1 anymore, cause we could be returning multiple StoreProducts for a single package)